ibis/expr/types/generic.py

```diff
@@ -1225,6 +1225,9 @@ def __pyarrow_result__(
         return data_mapper.convert_scalar(table[0][0], self.type())
 
     def __pandas_result__(self, df: pd.DataFrame) -> Any:
+        from ibis.formats.pandas import PandasData
+
+        df = PandasData.convert_table(df, ibis.schema({self.get_name(): self.type()}))
```

Comment:
There should be a `PandasData.convert_scalar()` method, I added that in one of my reverted commits, let me look for it.

Reply:
Here https://github.com/ibis-project/ibis/pull/7580/commits/ac0282c282459e22414de2ef96d02d8617b54751#diff-c3915f2476dac4d27022c405fd09c019b590114ef799dbddb3aab3a0c7c4b177L104-L108

Reply:
Could you please add that to the `PandasData` mapper?

Reply:
Yep!

Reply:
Done!